ARCHITECTURE.md

```diff
@@ -0,0 +1,103 @@
+- **Multiroom events**: through the use of the [cotejs](https://github.com/dashersw/cote) library and interfacing with the `audio` block, the `sound-supervisor` ensures that all devices on the same local network agree on which is the `master` device. To achieve this, `sound-supervisor` services on different devices exchange event messages constantly.
+- **API**: it creates a REST API on port 3000. The API allows other services to access current balenaSound configuration. This allows us to update configuration dynamically and have services react to it accordingly. As a general rule of thumb, if we are interested in service's configuration being able to be dynamically updated, they should rely on configuration reported by `sound-supervisor` and not on environment variables. At this moment, all services support this behaviour but their configuration is mostly static, you set it at startup via environment variables and that's it. However, there are *experimental* endpoints in the API to update configuration values and all services already support it. There's even a *secret* UI that allows for some configuration changes at runtime, it's located at `http://<DEVICE_IP>:3000/secret`.
+
+### Multi-room
+
+Multi-room services provide multiroom capabilities to balenaSound.
+
+**multiroom-server**
+This service runs a [Snapcast](https://github.com/badaix/snapcast) server which is responsible for broadcasting (and keeping it in sync) audio from the `audio` service into Snapcast clients. Clients can be running on the same device or on separate devices.
+
+**multiroom-client**
+Runs the client version of Snapcast. It needs to connect to a Snapcast server (can be a separate device) to receive audio packets. It will then forward the audio back into the `audio` service.
+
+### Plugins
+
+Plugins are the audio sources that generate the audio to be streamed/played. Refer to the plugins section below for pointers on how to add new plugins.
+
+## Audio routing
+
+Audio routing is the most cruicial part of balenaSound, it also changes significantly depending on what the current configuration is, with the biggest change being the mode of operation (multiroom vs standalone). There are two services controlling the audio routing:
+- the `audio` block is the key one as it's the one actually routing audio, so we'll zoom into it in sections below.
+- `sound-supervisor` on the other hand, is responsible for changing the routing according to what the current mode is. It will modify how sinks are internally connected depending on the mode of operation.
+
+**Note**: audio routing relies mainly on routing PulseAudio sinks. [Here](https://gavv.github.io/articles/pulseaudio-under-the-hood/) is an awesome resource on PulseAudio in case you are not familiar with it.
```

Comment:
```suggestion
**Note**: audio routing relies mainly on routing PulseAudio syncs. [Here](https://gavv.github.io/articles/pulseaudio-under-the-hood/) is an awesome resource on PulseAudio in case you are not familiar with it.
```